scene/main/node.h

```diff
@@ -170,6 +170,9 @@ class Node : public Object {
 		mutable LocalVector<Node *> children_cache;
 		HashMap<StringName, Node *> owned_unique_nodes;
 		bool unique_name_in_owner = false;
+		bool exposed_in_owner = false;
+		bool exposed_in_scene : 1;
+		bool exposed : 1;
```

Comment:
Feels like one of them is redundant. Wouldn't it be possible to infer `exposed` from `exposed_in_owner` and `exposed_in_scene`?

Reply:
I agree, the names are clunky and imply they overlap jobs, perhaps some better names/descriptions are in order.

1. **exposed_in_owner** : is the value saved in the .tscn and describes if the node has been exposed in its original scene. Useful to determine where you can expose/re-expose nodes
2.  **exposed_in_scene**: is used when saving the scene. determines whether the node is marked for exposure in the current scene (Writes out the [exposed path="...'] to the scene file if this is true)
3.  **exposed**:  is used on instantiation whether or not the node has been exposed in the instanced scene (Reads in the [exposed path="..."] from the scene file) Node will show up in instantiated scenes if this is true and allow you to re-expose them)

All of this is how we restrict exposed nodes to a single instantiation up, and not exposed forever like how @timoschwarzer suggested here https://github.com/godotengine/godot/pull/84018#issuecomment-2436382895

Thinking of renaming them to the following for clarity:
exposed_in_owner
marked_for_exposure
exposed_in_instance

Reply:
So isn't `exposed_in_owner` and `exposed` the same?

Reply:
I'll have to review the uses of exposed_in_owner vs exposed,  but you may be right on that, or that they at least serve the same purpose now...

Reply:
Found a use case/difference between the two.  Lets say we remove exposed_in_owner. 

Lets say we have 
- base.tscn where there is an exposed node
- scene_2.tscn that instantiates base.tscn and re-exposes that same node
- scene_3.tscn that instantiates scene_2.tscn

If we un-expose the node from base.tscn, and open scene_3.tscn  all we would know is that the node is exposed in scene_2.tscn so it would show that node as exposed, even though it no longer is in base.tscn. 

Now we could have something like <code>HashMap<StringName, Node *> owned_unique_nodes;</code> to store the exposed nodes and can look up if the owner of said node has it marked as exposed. But I feel that just moves where the same data is stored from the node to the owner. 

[example.zip](https://github.com/user-attachments/files/17608025/example.zip)

<details>
<summary>base.tscn</summary>

```diff
[gd_scene load_steps=2 format=3 uid="uid://comgrttev6gh3"]

[ext_resource type="Texture2D" uid="uid://dvj2xcm5vrvy8" path="res://icon.svg" id="1_ur6iq"]

[node name="base" type="Node2D"]

[node name="ExposedSprite" type="Sprite2D" parent="."]
exposed_in_owner = true
texture = ExtResource("1_ur6iq")

[exposed path="ExposedSprite"]
```
</details>

<details>
<summary>scene2.tscn</summary>

```diff
[gd_scene load_steps=2 format=3 uid="uid://um1kuqgfw87b"]

[ext_resource type="PackedScene" uid="uid://comgrttev6gh3" path="res://defend/base.tscn" id="1_lots5"]

[node name="scene_2" type="Node2D"]

[node name="base" parent="." instance=ExtResource("1_lots5")]

[exposed path="base/ExposedSprite"]
```
</details>

<details>
<summary>scene3.tscn</summary>

```cpp
[gd_scene load_steps=2 format=3 uid="uid://dm07442fbekfp"]

[ext_resource type="PackedScene" uid="uid://um1kuqgfw87b" path="res://defend/scene_2.tscn" id="1_w5frl"]

[node name="scene_3" type="Node2D"]

[node name="scene_2" parent="." instance=ExtResource("1_w5frl")]
```

</details>

```cpp
/* packed_scene.cpp:243   */
/* Where when instantiating a scene we expose nodes that scene specifies as exposed */
/* if it is exposed in the base scene */
if (p_edit_state == GEN_EDIT_STATE_MAIN || p_edit_state == GEN_EDIT_STATE_MAIN_INHERITED) {
	for (const NodePath &e_path : sdata->get_state()->exposed_nodes) {
		Node *ei = node->get_node_or_null(e_path);
		if (ei) {
			ei->set_exposed_in_instance(ei->is_exposed_in_owner());
		}
	}
}
```
Hopefully this all makes sense, I definitely confused myself a few times with the original names for those fields.

There is definitely room to reduce the use of is_exposed_in_owner where redundant, so will work on that